### PR TITLE
[New extension] Firefox history

### PR DESCRIPTION
**Pull request by cyprienruffino**
Re-implementing proposed in harindu95's PR albertlauncher/python/pull/21 as a native extension. The code mostly comes from the firefox bookmarks extension.

However, it seems to me that it would be better to include it in the "firefox bookmarks" extension since a lot of the code is actually redundant, even though the name might become unappropriated. Inheriting from FirefoxBookmarks::Extension is also impossible as extensions are final.

In a nutshell, the extension works, but I am unsatisfied of its actual status as a standalone extension. Please reject the PR if you consider that it should be part of firefox bookmarks.

### CODE REVIEW

**Comment by ManuelSchneid3r** [comment]
Well I hate disappointing ppl but I'd rather like to see this in the firefox[bookmarks] extension, to not piss of users with thousands of plugins. What do you think. Maybe idkcpp wants to join the discussion.

**Comment by cyprienruffino** [comment]
I also think that the proliferation of plugins is a bad idea, that's why I had doubts about creating a new one. The history feature could be a simple checkbox in the firefox bookmarks extension. Since most of the code is redundant anyways, it would simply enable a trigger and query the places.sqlite for history.

**Comment by ManuelSchneid3r** [comment]
Come to the chats and talk to @oe3umb or @idkCpp here. he is mainly maintaining the FF extension since I personally use chromium.

**Comment by idkCpp** [comment]
Yea, right. Here I am ^^'

I looked into the code and diff'ed it against the current FF extension. It appears that the SQL is the only change (apart from every "Bookmarks" and "History" change).
My approach would be to make the indexing function take the SQL as parameter and returns the index as result so that there is less duplication. I can look into it the coming weeks.

**Comment by ManuelSchneid3r** [comment]
Can you please merge this into the firefooxbookmarks extension and call it firefox?

**Comment by idkCpp** [comment]
Yes will do.

**Comment by ManuelSchneid3r** [comment]
Closing, since this will go into the ff ext

**Comment by ManuelSchneid3r** [comment]
@cyprienruffino you could go ahead if martin has no time.
